src/libs/actions/User.js

```diff
@@ -58,9 +61,35 @@ function changePassword(oldPassword, password) {
 }
 
 function getBetas() {
-    API.User_GetBetas().then((response) => {
+    return API.User_GetBetas().then((response) => {
         if (response.jsonCode === 200) {
             Onyx.set(ONYXKEYS.BETAS, response.betas);
+            return response.betas;
+        }
+        return [];
+    });
+}
+
+/**
+ * Fetches betas and User's preferred Locale, then set the app locale.
+ * @param {String} currentPreferredLocale
+ */
+function fetchBetasAndSetPreferredLocale(currentPreferredLocale) {
```

Comment:
I'm having trouble figuring out why we are combining these. It's not obvious from looking at the code so I don't think it's the right solution to the problem. 

If I'm understanding correctly, there is some race condition where we want to "set preferred locale" but only if we are on the beta and the betas have not loaded yet...?

This seems very easy to fix at the API layer by just returning the preferred locale _if_ the user is on the beta and do the check in the server. So basically have something like `API.User_GetPreferredLocale()` and then this would become:

```js
API.User_GetPreferredLocale()
    .then(({preferredLocale}) => {
        Onyx.set(ONYXKEYS.NVP_PREFERRED_LOCALE, preferredLocale);    
    });
```